src/mono/mono/mini/aot-compiler.c

```diff
@@ -9564,6 +9564,17 @@ append_mangled_klass (GString *s, MonoClass *klass)
 	return TRUE;
 }
 
+static const char*
+get_assembly_prefix (MonoImage *image)
+{
+	if (mono_is_corlib_image (image))
+		return "corlib";
+	else if (!strcmp (image->assembly->aname.name, "corlib"))
+		return "__corlib__";
+	else
+		return image->assembly->aname.name;
+}
```

Comment:
Why corlib only? Cannot we map the names into some indexes for all assemblies?

Reply:
This is just to make corlib llvm assembly more readable.